citizennet/purescript-ocelot
# AS-1146 Add Slider Component

## What does this pull request do?

Add slider component
- supports multiple values/thumbs
- allows changing the number of thumbs by query
- has two general modes depending on whether `marks` are specified
  - discrete (if `marks` are specified): thumbs can only sit on given set of values
  - continuous (if `marks` are left empty): thumbs can slide freely
- able to maintain minimal distance between thumbs if specified
- configurable layout (currently in limited form, see `Ocelot.Slider.Render.Config`)
- customizable render for intervals between thumbs

Features we may add in the future
- clicking on track teleports the closest thumb to the mouse position
- customizable render for marks
- indicator of thumb value when hovering over (especially in continuous setting)

## How should this be manually tested?

Kind of hard to put everything into words given there are enormous amount of edge cases. You have to play with it yourself a bit.

- [x] `make`
- [x] go to `dist/index.html#controls`
  - open console log to see current values
- [x] click on radio buttons to increase the number of thumbs
  - for example, click on `5` and spread the thumbs a bit and then click `6`
  - the new thumb will be added to a position that doesn't violate minimal distance constraint
- [x] click on radio buttons to lower the number of thumbs
  - extra thumbs should be removed from the right
- [x] have at least two thumbs and slide one against another
  - they can slide through each other but the minimal distance is always preserved
